### PR TITLE
Add dandiset version to landing page redirection

### PR DESCRIPTION
**Pull request by mgrauer**
This is to track #https://github.com/dandi/dandiarchive/pull/403, and shouldn't be merged until that PR is merged and deployed.

### CODE REVIEW

File: serve.py [hunk 1]
```diff
@@ -126,7 +126,7 @@ async def goto_dandiset_version(request, dataset, version):
         json_info = req.json()
         if json_info is not None:
             id = json_info['_id']
```
Comment (satra) [inline serve.py]: @mgrauer - will the girder endpoint return the right `_id` here?

alternatively it seems like we don't need to query girder at all and can pass the dataset identifier directly, which would affect both this and the previous route.

Comment (mgrauer) [reply serve.py]: I agree, we can just pass directly to the gui.

File: serve.py [hunk 2]
```diff
@@ -103,34 +102,23 @@ async def goto_public_dashboard(request):
...
-    if req.reason == 'OK':
-        json_info = req.json()
-        if json_info is not None:
-            id = json_info['_id']
-            url = f"https://gui.dandiarchive.org/#/dandiset/{id}"
-            if request.method == "HEAD":
-                return response.html(None, status=302, headers=make_header(url))
-            return response.redirect(url)
-    return response.text(f"dandi:{dataset:06d} not found.", status=404)
+    url = f"https://gui.dandiarchive.org/#/dandiset/{id}"
```
Comment (jjnesbitt) [inline serve.py]: Isn't the relevant variable `dataset`? 
```suggestion
    url = f"https://gui.dandiarchive.org/#/dandiset/{dataset:06d}"
```

**Comment by mgrauer** [comment]
Good catch, thanks. That commit suggestion feature on GH is cool, and new to me.

**Comment by satra** [comment]
@mgrauer - could you please rebase on master?

**Comment by mgrauer** [comment]
Done

**Pull request merged**